src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -0,0 +1,208 @@
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
+import java.lang.RuntimeException
+import java.security.MessageDigest
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+class UnionMangas(private val langOption: LanguageOption) : HttpSource() {
+    override val lang = langOption.lang
+
+    override val name: String = "Union Mangas"
+
+    override val baseUrl: String = "https://unionmangas.xyz"
+
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    val langApiInfix = when (lang) {
+        "it" -> langOption.infix
+        else -> "v3/po"
+    }
+
+    private fun apiHeaders(url: String): Headers {
+        val date = apiDateFormat.format(Date())
+        val path = url.toUrlWithoutDomain()
+
+        return headersBuilder()
+            .add("_hash", authorization(apiSeed, domain, date))
+            .add("_tranId", authorization(apiSeed, domain, date, path))
+            .add("_date", date)
+            .add("_domain", domain)
+            .add("_path", path)
+            .add("Origin", baseUrl)
+            .add("Host", apiUrl.removeProtocol())
+            .add("Referer", "$baseUrl/")
+            .build()
+    }
+
+    private fun authorization(vararg payloads: String): String {
+        val md = MessageDigest.getInstance("MD5")
+        val bytes = payloads.joinToString("").toByteArray()
+        val digest = md.digest(bytes)
+        return digest
+            .fold("") { str, byte -> str + "%02x".format(byte) }
+            .padStart(32, '0')
+    }
+
+    override fun chapterListParse(response: Response) = throw UnsupportedOperationException()
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapters = mutableListOf<SChapter>()
+        var currentPage = 0
+        do {
+            val chaptersDto = fetchChapterListPageable(manga, currentPage)
+            chapters += chaptersDto.toSChapter(langOption)
+            currentPage++
+        } while (chaptersDto.hasNextPage())
+        return Observable.just(chapters.reversed())
+    }
+
+    private fun fetchChapterListPageable(manga: SManga, page: Int): ChapterPageDto {
+        return try {
+            val maxResult = 16
+            val url = "$apiUrl/api/$langApiInfix/GetChapterListFilter/${manga.slug()}/$maxResult/$page/all/ASC"
+            return client
+                .newCall(GET(url, apiHeaders(url)))
+                .execute()
+                .parseAs<ChapterPageDto>()
+        } catch (e: RuntimeException) {
```

Comment:
I still don't understand why this exception needs to be caught.

Reply:
To take it clear, I'll remove the possible inconsistency when getting the chapters. I'll throw an error message to the user via Toast to take debugging easier in the future.

commit: https://github.com/keiyoushi/extensions-source/pull/2091/commits/f371e31b5d296d98f57895dcc59276ea1e086bd4

Reply:
if there is an error encountered that means the chapter parsing is broken in some way and the error should be thrown as is, no need to catch it